dart/lib/src/enricher/_web_enricher.dart

```diff
@@ -0,0 +1,96 @@
+import 'dart:async';
+
+import '../platform_checker.dart';
+
+import '../protocol.dart';
+import 'enricher.dart';
+import 'dart:html' as html show window, Window;
+
+final Enricher instance = WebEnricher(
+  html.window,
+  PlatformChecker(),
+);
+
+class WebEnricher implements Enricher {
+  WebEnricher(
+    this._window,
+    this._platformChecker,
+  );
+
+  final html.Window _window;
+  final PlatformChecker _platformChecker;
+
+  @override
+  FutureOr<SentryEvent> apply(
+      SentryEvent event, bool hasNativeIntegration) async {
+    // Web has no native integration, so no need to check for it
+
+    final contexts = event.contexts.copyWith(
+      device: await _getDevice(event.contexts.device),
+      operatingSystem: _getOperatingSystem(event.contexts.operatingSystem),
+      runtimes: _getRuntimes(event.contexts.runtimes),
+    );
+
+    return event.copyWith(
+      contexts: contexts,
+    );
+  }
+
+  Future<SentryDevice> _getDevice(SentryDevice? device) async {
+    return (device ?? SentryDevice()).copyWith(
+      online: _window.navigator.onLine,
+      memorySize: _getMemorySize(),
+      orientation: _getScreenOrientation(),
+      screenHeightPixels: _window.screen?.available.height.toInt(),
+      screenWidthPixels: _window.screen?.available.width.toInt(),
+      screenDensity: _window.devicePixelRatio.toDouble(),
+      timezone: DateTime.now().timeZoneName,
+    );
+  }
+
+  SentryOperatingSystem _getOperatingSystem(SentryOperatingSystem? os) {
+    return (os ?? SentryOperatingSystem()).copyWith(
+      name: _platformChecker.platform.operatingSystem,
+    );
+  }
+
+  List<SentryRuntime> _getRuntimes(List<SentryRuntime>? runtimes) {
+    // Pure Dart doesn't have specific runtimes per build mode
+    // like Flutter: https://flutter.dev/docs/testing/build-modes
+    final dartRuntime = SentryRuntime(
+      name: 'Dart',
+      rawDescription: 'Dart on browser',
+    );
+
+    final browserRuntime = SentryRuntime(
```

Comment:
I'm pretty sure this is not the best way to report it but I also don't know how the JavaScript SDK does it

Reply:
<img width="354" alt="Screenshot 2021-05-18 at 09 21 04" src="https://user-images.githubusercontent.com/5731772/118608799-6159b080-b7ba-11eb-8635-52878a6b1e3b.png">

just an example, pretty simple

Reply:
thats the browser though, for JS maybe it does not make sense to report a Runtime

Reply:
I guess we would then need to parse the User-Agent. Do we want to do that?

Reply:
I'd not parse but rather set it as it is, there's no standard format for user agent string